pkg/core/marshal/marshal.go

```diff
@@ -20,25 +20,47 @@ func UpdateNilFieldsWithZeroValue(i interface{}) {
 	}
 	numField := t.NumField()
 	for i := 0; i < numField; i++ {
-		zerovalueTag := t.Field(i).Tag.Get("default_zero_value")
-		if zerovalueTag != "true" {
-			continue
-		}
-
 		field := v.Field(i)
 		ft := t.Field(i)
-		if field.Kind() == reflect.Ptr {
-			ele := field.Elem()
+		updateNilFieldsWithZeroValue(ft.Type, field, ft.Tag)
+	}
+}
+
+func updateNilFieldsWithZeroValue(t reflect.Type, v reflect.Value, st reflect.StructTag) {
```

Comment:
Can you replace line 21 - 25 in `UpdateNilFieldsWithZeroValue` with a direct call to `updateNilFieldsWithZeroValue`? The logic seems to be identical to `reflect.Struct` case.

Reply:
Yes, you are right. Updated!